tests/conftest.py

```diff
@@ -144,6 +419,123 @@ def testing_samples(samples_collection_accessor):
+                connection.execute(roles_table.delete())
+                connection.execute(subjects_table.delete())
+                connection.execute(events_table.delete())
+                connection.execute(event_types_table.delete())
+                connection.execute(subject_types_table.delete())
+                connection.execute(role_types_table.delete())
+
+        delete_event_warehouse_data()
+        with event_wh_sql_engine.begin() as connection:
+            print("Inserting Events Warehouse test data")
+            connection.execute(role_types_table.insert(), EVENT_WH_DATA["role_types"])
+            connection.execute(event_types_table.insert(), EVENT_WH_DATA["event_types"])
+            connection.execute(subject_types_table.insert(), EVENT_WH_DATA["subject_types"])
+            connection.execute(subjects_table.insert(), EVENT_WH_DATA["subjects"])
+            connection.execute(events_table.insert(), EVENT_WH_DATA["events"])
+            connection.execute(roles_table.insert(), EVENT_WH_DATA["roles"])
+        yield
+    finally:
+        delete_event_warehouse_data()
+
+
+@pytest.fixture
+def mlwh_sample_stock_resource(config, mlwh_sql_engine):
+    # deletes
+    delete_from_mlwh(
+        MLWH_SAMPLE_STOCK_RESOURCE["stock_resource"],
+        mlwh_sql_engine,
+        config.MLWH_STOCK_RESOURCES_TABLE,
+    )
+    delete_from_mlwh(MLWH_SAMPLE_STOCK_RESOURCE["sample"], mlwh_sql_engine, config.MLWH_SAMPLE_TABLE)
+    delete_from_mlwh(MLWH_SAMPLE_STOCK_RESOURCE["study"], mlwh_sql_engine, config.MLWH_STUDY_TABLE)
+
+    # inserts
+    insert_into_mlwh(MLWH_SAMPLE_STOCK_RESOURCE["sample"], mlwh_sql_engine, config.MLWH_SAMPLE_TABLE)
+    insert_into_mlwh(MLWH_SAMPLE_STOCK_RESOURCE["study"], mlwh_sql_engine, config.MLWH_STUDY_TABLE)
+    insert_into_mlwh(
+        MLWH_SAMPLE_STOCK_RESOURCE["stock_resource"],
+        mlwh_sql_engine,
+        config.MLWH_STOCK_RESOURCES_TABLE,
+    )
+
+
+def insert_into_mlwh(data, mlwh_sql_engine, table_name):
+    table = get_table(mlwh_sql_engine, table_name)
+
+    with mlwh_sql_engine.begin() as connection:
+        connection.execute(table.delete())  # delete all rows from table first
+        print("Inserting MLWH test data")
+        connection.execute(table.insert(), data)
+
+
+def delete_from_mlwh(data, mlwh_sql_engine, table_name):
+    table = get_table(mlwh_sql_engine, table_name)
+
+    with mlwh_sql_engine.begin() as connection:
+        print("Deleting MLWH test data")
+        connection.execute(table.delete())
+
+
+def get_table(sql_engine: Engine, table_name: str):
+    metadata = MetaData(sql_engine)
+    metadata.reflect()
+    return metadata.tables[table_name]
+
+
+@pytest.fixture
+def event_wh_sql_engine(config):
+    create_engine_string = f"mysql+pymysql://{config.WAREHOUSES_RW_CONN_STRING}/{config.EVENTS_WH_DB}"
+    return sqlalchemy.create_engine(create_engine_string, pool_recycle=3600)
+
+
+@pytest.fixture
+def mlwh_sql_engine(config):
```

Comment:
As mentioned, not sure if `event_wh_sql_engine` and `mlwh_sql_engine` need to be test fixtures. Could be wrong though

Reply:
As discussed in a previous comment, keeping these as test fixtures